scalding-core/src/main/scala/com/twitter/scalding/RichPipe.scala

```diff
@@ -53,6 +54,39 @@ object RichPipe extends java.io.Serializable {
     }
     p
   }
+
+  // A pipe can have more than one description when merged together, so we store them delimited with 255.toChar.
+  // Cannot use 1.toChar as we get an error if it is not a printable character.
+  private def encodePipeDescriptions(descriptions: Seq[String]): String = {
+    descriptions.map(_.replace(255.toChar, ' ')).filter(_.nonEmpty).mkString(255.toChar.toString)
+  }
+
+  private def decodePipeDescriptions(encoding: String): Seq[String] = {
+    encoding.split(255.toChar).toSeq
+  }
+
+  def getPipeDescriptions(p: Pipe): Seq[String] = {
+    if (p.getStepConfigDef.isEmpty)
+      Nil
+    else {
+      val encodedResult = p.getStepConfigDef.apply(Config.PipeDescriptions, new Getter {
+        override def update(s: String, s1: String): String = ???
+        override def get(s: String): String = null
+      })
+      Some(encodedResult)
```

Comment:
Option(encodedResult)
 .filterNot(_.isEmpty)

might be simpler. But no biggie if you don't want to do this.